nomad/node_endpoint.go

```diff
@@ -944,7 +944,9 @@ func (n *Node) DeriveVaultToken(args *structs.DeriveVaultTokenRequest,
 	// setErr is a helper for setting the recoverable error on the reply and
 	// logging it
 	setErr := func(e error, recoverable bool) {
-		reply.Error = structs.NewRecoverableError(e, recoverable).(*structs.RecoverableError)
+		if rerr, ok := structs.NewRecoverableError(e, recoverable).(*structs.RecoverableError); ok {
```

Comment:
This approach will still log an error even when `e` is nil.

Instead I would just check if `e` is nil and return early:
```go
setErr := func(...) {
    if e == nil { return }

    // ...existing code...
}
```